test/Main.hs

```diff
@@ -73,6 +82,58 @@ spec = do
         dpiTimeStampToUTCDPITimeStamp dpi `shouldBe` expected
 
       it "Should roundtrip UTCTime through DPITimestamp (w/ nanos -- not picos) " $ \_ -> do
-        property $ \tod day (nanos :: Nano) -> do
-          let utc = UTCTime day $ timeOfDayToTime tod { todSec = realToFrac nanos }
+        property $ \(UTCTimeNanos utc) -> do
           utc `shouldBe` dpiTimeStampToUTCTime (utcTimeToDPITimestamp utc)
+
+    describe "Roundtrip tests" $ do
+      it "Should round trip random values from a table" $ \conn -> do
+        property $ \x@TestTable{..} -> do
+          execute_ conn "create table test (a varchar(300), b number (12,0), c number (12,0), d number (12,0) null, e timestamp, f number (38,28))"
```

Comment:
This test can be fixed by specifying a precision for the timestamp field:
```sql
create table test (e timestamp(9))
```
The precision applies to the fsecond field, and is set to 6 by default.

Reply:
Updated